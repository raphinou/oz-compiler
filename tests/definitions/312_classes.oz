% Attribute assignation with <-
local
   C I
in
   class C
      attr
         count:0
      meth init skip end
      meth echo(S)
         {Show S}
         count<-@count+1
      end
      meth stats
         {Show @count}
      end
   end
   I={New C init}
   {I echo(1)}
   {I echo(a)}
   {I echo(z)}
   {I stats}

end

