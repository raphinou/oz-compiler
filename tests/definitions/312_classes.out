1
a
z
3
